R/prep_diagram.R

```diff
@@ -2,5 +2,67 @@ prep_diagram <- function(audit_data = NULL, emissions_data = NULL) {
   if (is.null(audit_data) | is.null(emissions_data)) {
     data_out <- use_toy_data("diagram")
   }
-  data_out
+
+  audit_data <- audit_data %>%
+    dplyr::mutate(exposure_portfolio = sum(.data$value_usd, na.rm = TRUE)) %>%
+    dplyr::group_by(.data$asset_type) %>%
+    dplyr::mutate(
+      exposure_asset_class = sum(.data$value_usd, na.rm = TRUE),
+      exposure_asset_class_perc = .data$exposure_asset_class / .data$exposure_portfolio
+    ) %>%
+    dplyr::ungroup() %>%
+    dplyr::mutate(
+      covered_pacta = dplyr::if_else(
+        .data$financial_sector %in% c("Automotive", "Coal", "Oil&Gas", "Power", "Aviation", "Cement", "Steel"),
+        TRUE,
+        FALSE
+      )
+    ) %>%
+    dplyr::group_by(
+      .data$asset_type, .data$exposure_portfolio, .data$covered_pacta,
+      .data$exposure_asset_class, .data$exposure_asset_class_perc
+    ) %>%
+    dplyr::summarise(
+      exposure_pacta = sum(.data$value_usd, na.rm = TRUE),
+      .groups = "drop"
+    ) %>%
+    dplyr::mutate(
+      exposure_pacta_perc_asset_class_exposure = .data$exposure_pacta / .data$exposure_asset_class
+    ) %>%
+    dplyr::ungroup() %>%
+    dplyr::filter(.data$covered_pacta) %>%
+    dplyr::select(
+      c(exposure_portfolio, asset_type, exposure_asset_class,
+        exposure_asset_class_perc, exposure_pacta, exposure_pacta_perc_asset_class_exposure)
+    )
```

Comment:
```suggestion
    dplyr::select(
      c("exposure_portfolio", "asset_type", "exposure_asset_class",
        "exposure_asset_class_perc", "exposure_pacta", "exposure_pacta_perc_asset_class_exposure")
    )
```

Reply:
quotes so these do not generate notes in R CMD check about non-visible global variables